agent/structs/config_entry_discoverychain.go

```diff
@@ -471,6 +471,9 @@ type ServiceRouteDestination struct {
 	// Allow HTTP header manipulation to be configured.
 	RequestHeaders  *HTTPHeaderModifiers `json:",omitempty" alias:"request_headers"`
 	ResponseHeaders *HTTPHeaderModifiers `json:",omitempty" alias:"response_headers"`
+
+	// RequestRedirect TODO: Melisa update this comment
+	RequestRedirect *RequestRedirect `json:",omitempty" alias:"request_redirect"`
```

Comment:
Validation-wise you're going to want it to be mutually exclusive with: namespace, partition, service, serviceSubset